ci-operator/step-registry/cucushift/installer/check/gcp/confidential-computing/cucushift-installer-check-gcp-confidential-computing-commands.sh

```diff
@@ -23,39 +21,117 @@ then
   gcloud config set project "${GOOGLE_PROJECT_ID}"
 fi
 
-## Try the validation
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
 ret=0
 
-echo "$(date -u --rfc-3339=seconds) - Checking Confidential Computing settings of cluster machines..."
-readarray -t machines < <(gcloud compute instances list --filter="name~${CLUSTER_NAME}" --format="table(name,zone)" | grep -v NAME)
-for line in "${machines[@]}"; do
-  machine_name="${line%% *}"
-  machine_zone="${line##* }"
-  gcloud compute instances describe "${machine_name}" --zone "${machine_zone}" --format json > "/tmp/${CLUSTER_NAME}-machine.json"
-  confidential_compute="$(jq -r -c .confidentialInstanceConfig.enableConfidentialCompute "/tmp/${CLUSTER_NAME}-machine.json")"
-  on_host_maintenance="$(jq -r -c .scheduling.onHostMaintenance "/tmp/${CLUSTER_NAME}-machine.json")"
-  cpu_platform="$(jq -r -c .cpuPlatform "/tmp/${CLUSTER_NAME}-machine.json")"
-
-  if [[ "${confidential_compute}" == true ]]; then
-    echo "$(date -u --rfc-3339=seconds) - Matched .enableConfidentialCompute '${confidential_compute}' for '${machine_name}'."
-  else
-    echo "$(date -u --rfc-3339=seconds) - Unexpected .enableConfidentialCompute '${confidential_compute}' for '${machine_name}'."
-    ret=1
+control_plane_node_cc_type=""
+compute_node_cc_type=""
+control_plane_node_maintenance_policy=""
+compute_node_maintenance_policy=""
+if [[ -n "${CONFIDENTIAL_COMPUTE}" ]]; then
+  control_plane_node_cc_type="${CONFIDENTIAL_COMPUTE}"
+  compute_node_cc_type="${CONFIDENTIAL_COMPUTE}"
+fi
+if [[ -n "${ON_HOST_MAINTENANCE}" ]]; then
+  control_plane_node_maintenance_policy="${ON_HOST_MAINTENANCE}"
+  compute_node_maintenance_policy="${ON_HOST_MAINTENANCE}"
+fi
+if [[ -n "${CONTROL_PLANE_CONFIDENTIAL_COMPUTE}" ]]; then
+  control_plane_node_cc_type="${CONTROL_PLANE_CONFIDENTIAL_COMPUTE}"
+fi
+if [[ -n "${CONTROL_PLANE_ON_HOST_MAINTENANCE}" ]]; then
+  control_plane_node_maintenance_policy="${CONTROL_PLANE_ON_HOST_MAINTENANCE}"
+fi
+if [[ -n "${COMPUTE_CONFIDENTIAL_COMPUTE}" ]]; then
+  compute_node_cc_type="${COMPUTE_CONFIDENTIAL_COMPUTE}"
+fi
+if [[ -n "${COMPUTE_ON_HOST_MAINTENANCE}" ]]; then
+  compute_node_maintenance_policy="${COMPUTE_ON_HOST_MAINTENANCE}"
+fi
+
+declare -A cc_type_mapping
+cc_type_mapping["Enabled"]="true"
+cc_type_mapping["AMDEncryptedVirtualization"]="SEV"
+cc_type_mapping["AMDEncryptedVirtualizationNestedPaging"]="SEV_SNP"
+cc_type_mapping["IntelTrustedDomainExtensions"]="TDX"
+
+echo "$(date -u --rfc-3339=seconds) - INFO: Checking the Confidential Computing settings of the control-plane machines..."
+oc get nodes -l node-role.kubernetes.io/master -o json > /tmp/cluster_nodes.json
+for line in $(jq '.items[].spec.providerID' "/tmp/cluster_nodes.json"); do
+  machine_name=$(echo "${line}" | cut -d\/ -f5)
+  machine_name="${machine_name%\"}"
```

Comment:
I wonder the machine name got from `.items[].metadata.name` is not reliable ?

Reply:
@jianlinliu Sorry, not quite understand. Could you please clarify a bit? Thanks!

Reply:
I found the value from `oc get nodes -l node-role.kubernetes.io/master -o json | jq -r '.items[].metadata.name'`  is the machine name, I wonder why we are query `.items[].spec.providerID` to get the machine name, is there any difference or advantage ?

Reply:
FYI I am using `.items[].spec.providerID` instead to get both the machine name and the machine zone, which are required for the command `gcloud compute instances describe`. That's it, thanks!